[Tan Jia Xian] iP

tanjiaxian99 opened the pull request:
# Duke Chatbot: Lollipop

> v0.1 can be found [here](https://github.com/tanjiaxian99/ip/releases/tag/v0.1)

**Lollipop does the following things:**
- Add different tasks, such as *Todos*, *Deadlines* and *Events*
- Delete tasks
- Mark tasks as done

**Things that have been accomplished**:
1. Everything up to level 9
2. Push, commit, branch, PR
3. Writing GFMD

**New things to do:**
- [X] ~~Write GFMD~~ 👍
- [ ] Review some peer PRs
- [ ] Learn from others (optional)
- [ ] Add increments as branches

**An example code block found in the `main` class:**
```java
/** Prints the welcome message */
public void printWelcome() {
    System.out.println(
            "Lollipop: Hello! I am your personal chatbot, Lollipop! "
            + "What would you like to do today?");
}
```

Commit:
Add Gradle support

tanjiaxian99 committed:
updated gitignore

tanjiaxian99 committed:
added greet echo exit

tanjiaxian99 committed:
added commands for adding task and retrieving task list

tanjiaxian99 committed:
added Task class

tanjiaxian99 committed:
added command mark as done and added getDescription to Task class

tanjiaxian99 committed:
added classes Todo, Deadline and Event

tanjiaxian99 committed:
added commands todo, deadline, event

tanjiaxian99 committed:
updated runtest.bat, input.txt and EXPECTED.txt

tanjiaxian99 committed:
added DukeException class

tanjiaxian99 committed:
added exceptions for missing descriptions, keywords, invalid task number and invalid number format

tanjiaxian99 committed:
added delete command

tanjiaxian99 committed:
updated input.txt and EXPECTED.txt

tanjiaxian99 committed:
Add TestFile class which opens ./data/data.txt

tanjiaxian99 committed:
Change updateTextFile to static method and update the file through commands

tanjiaxian99 committed:
Change formatting of tasks in task_list.txt

tanjiaxian99 committed:
Add readFromTextFile

tanjiaxian99 committed:
Change readFromTextFile to parse text and update taskList

tanjiaxian99 committed:
Update error message when file cannot be found

tanjiaxian99 committed:
Update Deadline to take in a date

tanjiaxian99 committed:
Update Event to take in a date

tanjiaxian99 committed:
Add date command which prints deadlines/events that occur on the given date

tanjiaxian99 committed:
Merge branch 'branch-Level-7'

tanjiaxian99 committed:
Merge branch 'branch-Level-8'

tanjiaxian99 committed:
Remove data/task_list.txt

tanjiaxian99 committed:
Add new class UI and moved UI methods to UI class

tanjiaxian99 committed:
Remove TextFile and add Storage

tanjiaxian99 committed:
Add TaskList class and shift tasks manipulation methods

tanjiaxian99 committed:
Add Command classes and Parser class

tanjiaxian99 committed:
Split files into different packages

tanjiaxian99 committed:
Add tests for ByeCommand, Deadline, Event

tanjiaxian99 committed:
Update META-INF

tanjiaxian99 committed:
Add JavaDoc for all public methods

tanjiaxian99 committed:
Tweak code to comply with coding standard

tanjiaxian99 committed:
Add Find command

tanjiaxian99 committed:
Merge branch 'branch-A-JavaDoc'

tanjiaxian99 committed:
Merge branch 'branch-A-CodingStandard'

tanjiaxian99 committed:
Merge branch 'branch-Level-9'

tanjiaxian99 committed:
Encapsulate logic of printing methods to TaskList

tanjiaxian99 committed:
Merge branch 'add-gradle-support' of github.com:tanjiaxian99/ip into branch-A-Gradle

tanjiaxian99 committed:
Add Gradle support

tanjiaxian99 committed:
Merge branch 'branch-A-Gradle'

tanjiaxian99 committed:
Add checkstyle.xml and suppressions.xml and update code based on checkstyle

tanjiaxian99 committed:
Merge branch 'branch-A-Checkstyle'

tanjiaxian99 committed:
Refactor code to include JavaFX

tanjiaxian99 committed:
Add first message

tanjiaxian99 committed:
Add handleExitProgram while closes application when user types Bye

tanjiaxian99 committed:
Create data_list.txt if the file does not exist

tanjiaxian99 committed:
Change avatar images

tanjiaxian99 committed:
Add padding and spacing in HBox

tanjiaxian99 committed:
Adjust code to abide by coding standard

tanjiaxian99 committed:
Merge branch 'branch-Level-10'

pbthang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,106 @@
+            String description = splitCommand[1];
+            String[] splitDescription = description.split(" /by ");
+            if (splitDescription.length == 1) {
+                throw new DukeException("Please add in /by, following by a dateline.");
```
Minor typo. Should it be `deadline` instead of `dateline`?

pbthang reviewed (commented):
Sorry, wrong PR :))

pualixue commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,8 @@
+package duke;
+
+public class DukeException extends Exception {
```
Perhaps you would like to add a Javadoc comment for this class?

pualixue commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,8 @@
+
+public class DukeException extends Exception {
+
+    public DukeException(String message) {
```
Perhaps you would like to add a Javadoc comment for this public constructor?

pualixue commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,106 @@
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+
+public class Parser {
```
Perhaps you would like to add a Javadoc comment for this class?

pualixue commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,106 @@
+     * @return A command object that can be executed.
+     * @throws DukeException If the command is invalid
+     */
+    public static Command parse(String command) throws DukeException {
```
This method seems a bit long as stated by the [code quality](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods).
Maybe the codes under different cases (eg. `command.startsWith("deadline")`, `command.startsWith("event")` etc.) can each be extracted out into a private method?

tanjiaxian99 replied:
I really like the idea of abstracting out the different cases into their own separate methods. Thank you for your suggestion!

pualixue reviewed (commented):
LGTM. Just some missing Javadoc comments for some classes.

huisthat commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,18 @@
+import duke.task.TaskList;
+
+public class ByeCommand extends Command {
+
```
might be good to add a javadocs comment for this class and its methods?

huisthat commented on src/main/java/duke/command/DateCommand.java:
```diff
@@ -0,0 +1,19 @@
+import duke.ResponseLogic;
+import duke.Storage;
+import duke.task.TaskList;
+
```
might be good to add a javadocs comment for this class and its methods as well?

huisthat commented on src/main/java/duke/command/DateCommand.java:
```diff
@@ -0,0 +1,19 @@
+public class DateCommand extends Command {
+
+    private String date;
+
```
might be good to add a javadocs comment for this class and its methods as well?

huisthat commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,13 @@
+package duke.task;
+
+public class Todo extends Task {
+
```
might be good to add a javadocs comment for this class and its methods as well?

tanjiaxian99 replied:
Thank you for your comments! I will add Javadocs to all my classes and methods.

huisthat reviewed (commented):
looks good!

tanjiaxian99 committed:
Add assertions

Add assertions to check for control-flow invariants, as well as to
verify that initial variables like "tasks" is not null.

This will help to minimise potential errors in the future, where
tasks is not declared, or is neither a Todo, Deadline or Event.

tanjiaxian99 committed:
Update JavaDocs

Add comments to classes, constructors and some methods to abide to
the Java coding standard. This will allow future developers to be
clear on what each class and method does.

tanjiaxian99 committed:
Abstract parse() method into individual get methods.

The previous parse() method spanned a few hundred lines and was
incredibly hard to read. To improve code quality, each block within
if-else now has dedicated get method for each command.

SLAP has been applied to the parse() method so that each block is
abstracted away at the same level, maintaining readbability. In future
code revisions, developers would only have to modify the private get
methods to change the functionality of the parse() method.

tanjiaxian99 committed:
Add this. to properties

tanjiaxian99 committed:
Merge pull request #1 from tanjiaxian99/branch-A-Assertions

Add assertions

tanjiaxian99 committed:
Merge master

tanjiaxian99 committed:
Merge pull request #2 from tanjiaxian99/branch-A-CodeQuality

Improve code quality

tanjiaxian99 committed:
Give aliases for commands

tanjiaxian99 committed:
Merge branch 'branch-C-FriendlierSyntax'

tanjiaxian99 committed:
Fix program not closing when 'q' is sent

tanjiaxian99 committed:
Remove background from Lollipop avatar

tanjiaxian99 committed:
Change response for indexOutOfBoundsException

tanjiaxian99 committed:
Merge branch 'branch-A-BetterGui'

tanjiaxian99 committed:
Update user icon and acknowledge authors in README

tanjiaxian99 committed:
Add icons source to welcome message

tanjiaxian99 committed:
Add title to window

tanjiaxian99 committed:
Add user guide

tanjiaxian99 committed:
Add 'q' keyword to user guide

tanjiaxian99 committed:
Remove assert statement from getTasksOnDate()

tanjiaxian99 committed:
Set theme jekyll-theme-cayman

tanjiaxian99 committed:
Merge branch 'master' of github.com:tanjiaxian99/ip

tanjiaxian99 committed:
Made minor changes to the user guide

tanjiaxian99 committed:
Update header comments
